client/state/analytics/actions.js

```diff
@@ -50,12 +57,25 @@ export const recordEvent = ( service, args ) => ( {
 	}
 } );
 
+export const recordEventOnce = ( eventId, service, args ) => ( dispatch, getState ) => {
+	const state = getState();
```

Comment:
In general, I think we should never use `getState` in thunks since it makes the action creator magical and less predictable. Instead, I wonder if we should just dispatch the `recordEvent` with `eventId` as an added `args`, then determine from the [middleware](https://github.com/Automattic/wp-calypso/blob/b6d0d27438a16cf7c8700cf4ed8b70dbc42805e3/client/state/analytics/middleware.js#L25-L36) whether to skip invoking the event tracking. This _could_ even be a module-level variable tracking already-tracked `eventId` (since analytics middleware is only run in browser), though makes it difficult to port to the server if that's an eventual goal. Otherwise we could keep tracked events in state (as you've already done here) and access from the [state object passed to middleware](http://redux.js.org/docs/api/applyMiddleware.html#arguments).